Source/WebKit/UIProcess/UserMediaPermissionRequestManagerProxy.cpp

```diff
@@ -342,25 +356,32 @@ void UserMediaPermissionRequestManagerProxy::finishGrantingRequest(UserMediaPerm
         if (!page)
             return;
 
+        auto* frame = WebFrameProxy::webFrame(request->frameID());
+        if (!frame)
+            return;
+        Ref process = frame->process();
+
         ++m_hasPendingCapture;
 
         Vector<SandboxExtension::Handle> handles;
 #if PLATFORM(COCOA)
-        if (!m_hasCreatedSandboxExtensionForTCCD && doesPageNeedTCCD(*page)) {
-            handles = SandboxExtension::createHandlesForMachLookup({ "com.apple.tccd"_s }, page->legacyMainFrameProcess().auditToken(), SandboxExtension::MachBootstrapOptions::EnableMachBootstrap);
-            m_hasCreatedSandboxExtensionForTCCD = true;
+        if (!m_hasCreatedSandboxExtensionForTCCD.contains(process->coreProcessIdentifier()) && doesPageNeedTCCD(*page)) {
+            handles = SandboxExtension::createHandlesForMachLookup({ "com.apple.tccd"_s }, process->auditToken(), SandboxExtension::MachBootstrapOptions::EnableMachBootstrap);
+            m_hasCreatedSandboxExtensionForTCCD.add(process->coreProcessIdentifier());
```

Comment:
Maybe `m_hasCreatedSandboxExtensionForTCCD` should be a `WeakHashSet<WebProcessProxy>`, so that its growth is bounded.